module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -82,14 +103,35 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $localDirs = defined('LOCAL_OVERRIDE_DIR')
-            && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
-                ? [
-                    [
-                        'directory' => LOCAL_OVERRIDE_DIR,
-                        'defaultConfigSubdir' => $this->defaultLocalConfigSubdir,
-                    ],
-                ] : [];
+        $localDirs = [];
+        $currentDir = defined('LOCAL_OVERRIDE_DIR')
+            && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0 ?
+            LOCAL_OVERRIDE_DIR : '';
+        while (!empty($currentDir)) {
+            $systemConfigFile = $currentDir . '/localSystem.ini';
+            $systemConfig = new Config(
+                file_exists($systemConfigFile) ?
+                    $this->iniReader->fromFile($systemConfigFile) :
+                    []
+            );
+            array_unshift(
+                $localDirs,
+                [
+                    'directory' => $currentDir,
+                    'defaultConfigSubdir' =>
+                        $systemConfig['Local_Dir']['config_subdir'] ??
+                        $this->defaultLocalConfigSubdir,
+                ]
+            );
+            $parentDir = '';
+            if (!empty($systemConfig['Parent_Dir']['absolute_path'])) {
+                $parentDir = $systemConfig['Parent_Dir']['absolute_path'];
+            }
+            if (empty($parentDir) && !empty($systemConfig['Parent_Dir']['relative_path'])) {
+                $parentDir = $currentDir . '/' . $systemConfig['Parent_Dir']['relative_path'];
```

Comment:
Should we throw some kind of an exception if both absolute_path and relative_path are set? Or should we adjust the configuration to make such a conflict impossible -- e.g. by having a path setting and a path_type setting that specifies 'relative' or 'absolute'?

Reply:
I used the Parent_Config logic as orientation, but I think you are right. It should not be possible contradicting configs.

I changed it into "path" and a boolean "is_relative_path" to define whether the path is relative or absolute.